src/dotplot.jl

```diff
@@ -0,0 +1,183 @@
+
+# ---------------------------------------------------------------------------
+# Dot Plot (strip plot, beeswarm)
+
+@recipe function f(::Type{Val{:dotplot}}, x, y, z; mode = :none, side=:both, dy = 0.1)
+    # if only y is provided, then x will be UnitRange 1:length(y)
+    if typeof(x) <: AbstractRange
+        if step(x) == first(x) == 1
+            x = plotattributes[:series_plotindex]
+        else
+            x = [getindex(x, plotattributes[:series_plotindex])]
+        end
+
+        x = repeat([x], length(y))
+    end
+    x = Float64.(x)
```

Comment:
```julia
df = DataFrame(x = repeat(["A", "B"], outer = 100), y = randn(200))
@df df dotplot(:x, :y)
```
errors here:
```
ERROR: MethodError: no method matching Float64(::String)
Closest candidates are:
  Float64(::Int8) at float.jl:60
  Float64(::Int16) at float.jl:60
  Float64(::Int32) at float.jl:60
  ...
Stacktrace:
 [1] _broadcast_getindex_evalf at .\broadcast.jl:578 [inlined]
 [2] _broadcast_getindex at .\broadcast.jl:551 [inlined]
 [3] getindex at .\broadcast.jl:511 [inlined]
 [4] copy at .\broadcast.jl:787 [inlined]
 [5] materialize(::Base.Broadcast.Broadcasted{Base.Broadcast.DefaultArrayStyle{1},Nothing,Type{Float64},Tuple{Array{String,1}}}) at .\broadcast.jl:753
 [6] macro expansion at C:\Users\Daniel\.julia\dev\StatsPlots\src\dotplot.jl:16 [inlined]
 [7] apply_recipe(::Dict{Symbol,Any}, ::Type{Val{:dotplot}}, ::Array{String,1}, ::Array{Float64,1}, ::Nothing) at C:\Users\Daniel\.julia\packages\RecipesBase\OPemU\src\RecipesBase.jl:275
 [8] _process_seriesrecipe(::Plots.Plot{Plots.GRBackend}, ::Dict{Symbol,Any}) at C:\Users\Daniel\.julia\dev\Plots\src\pipeline.jl:408
 [9] _plot!(::Plots.Plot{Plots.GRBackend}, ::Dict{Symbol,Any}, ::Tuple{Array{String,1},Array{Float64,1}}) at C:\Users\Daniel\.julia\dev\Plots\src\plot.jl:234
 [10] #plot#133(::Base.Iterators.Pairs{Symbol,Symbol,Tuple{Symbol},NamedTuple{(:seriestype,),Tuple{Symbol}}}, ::Function, ::Array{String,1}, ::Vararg{Any,N} where N) at C:\Users\Daniel\.julia\dev\Plots\src\plot.jl:57
 [11] (::getfield(RecipesBase, Symbol("#kw##plot")))(::NamedTuple{(:seriestype,),Tuple{Symbol}}, ::typeof(plot), ::Array{String,1}, ::Array{Float64,1}) at .\none:0
 [12] #dotplot#103(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Array{String,1}, ::Vararg{Any,N} where N) at C:\Users\Daniel\.julia\packages\RecipesBase\OPemU\src\RecipesBase.jl:369
 [13] dotplot(::Array{String,1}, ::Vararg{Any,N} where N) at C:\Users\Daniel\.julia\packages\RecipesBase\OPemU\src\RecipesBase.jl:369
 [14] #add_label#17(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Array{String,1}, ::Function, ::Array{String,1},
::Vararg{Any,N} where N) at C:\Users\Daniel\.julia\dev\StatsPlots\src\df.jl:139
 [15] add_label(::Array{String,1}, ::Function, ::Array{String,1}, ::Vararg{Any,N} where N) at C:\Users\Daniel\.julia\dev\StatsPlots\src\df.jl:137
 [16] (::getfield(Main, Symbol("##51#52")))(::DataFrame) at .\none:0
 [17] top-level scope at none:0
```
`@df df violin(:x, :y)` works, if you want to look that recipe is implemented.

Reply:
Yeah, the cause was a quick hack to get around something I saw. Thanks for pointing out it is a problem for df.